internal/server/iifo.go

```diff
@@ -15,3 +22,58 @@ func (s *SOMASServer) runIIFOEndOfTurn() error {
 	// TODO:- IIFO team
 	return nil
 }
+
+func (s *SOMASServer) runPredictionSession() error {
+	s.logf("start runPredictionSession")
+	islandPredictionDict, err := s.getPredictions()
+	if err != nil {
+		return err
+	}
+
+	err = s.distributePredictions(islandPredictionDict)
+	if err != nil {
+		return err
+	}
+
+	defer s.logf("finish runPredictionSession")
+	return nil
+}
+
+func (s *SOMASServer) getPredictions() (shared.PredictionInfoDict, error) {
+	islandPredictionsDict := shared.PredictionInfoDict{}
+	var err error
+	for id, client := range s.clientMap {
+		islandPredictionsDict[id], err = client.MakePrediction()
+		if err != nil {
+			return islandPredictionsDict, err
+		}
+	}
+	return islandPredictionsDict, nil
+}
+func (s *SOMASServer) distributePredictions(islandPredictionDict shared.PredictionInfoDict) error {
+	recievedPredictionsDict := make(shared.RecievedPredictionsDict)
+	var err error
+	// Add the predictions/sources to the dict containing which predictions each island should recieve
+	// Don't allow teams to know who else these predictions were shared with in MVP
+	for idSource, info := range islandPredictionDict {
+		for _, idShare := range info.TeamsOfferedTo {
+			if idShare == idSource {
+				continue
+			}
+			if recievedPredictionsDict[idShare] == nil {
+				recievedPredictionsDict[idShare] = make(shared.PredictionInfoDict)
+			}
+			info.TeamsOfferedTo = nil
+			recievedPredictionsDict[idShare][idSource] = info
+		}
+	}
+
+	// Now distribute these predictions to the islands
+	for id, client := range s.clientMap {
+		err = client.RecievePredictions(recievedPredictionsDict[id])
+		if err != nil {
+			return err

```

Comment:
Should wrap this error as well, client may return some bogus error that is hard to track.

Reply:
We love bogus errors!